iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -469,12 +473,29 @@ private Long getSnapshotId(Table table, String branchName) {
   /**
    * Creates and commits an Iceberg change with the provided data and delete files.
    * If there are no delete files then an Iceberg 'append' is created, otherwise Iceberg 'overwrite' is created.
-   * @param table The table we are changing
-   * @param startTime The start time of the commit - used only for logging
-   * @param results The object containing the new files we would like to add to the table
+   *
+   * @param table        The table we are changing
+   * @param startTime    The start time of the commit - used only for logging
+   * @param results      The object containing the new files we would like to add to the table
+   * @param isCompaction Flag that indicates whether or not the operation is done as part of compaction
    */
   private void commitWrite(Table table, String branchName, Long snapshotId, long startTime,
-      FilesForCommit results, Operation operation) {
+                           FilesForCommit results, Operation operation, boolean isCompaction) {
+
+    if (isCompaction) {
```

Comment:
If i want to specify the output data file size using iceberg property `target-file-size-bytes`, how to do?
Can we refer to https://iceberg.apache.org/docs/latest/spark-procedures/#rewrite_data_files to allow users to specify property to rewrite data files?

Reply:
@difin during iceberg table write, there must be some configuration which will be deciding the file size for normal data writes. So, if you set that conf, you should be able to achieve what is configured as part of ``target-file-size-bytes``?

So, the new files would be written of size mentioned as part of ``target-file-size-bytes``?

This could be 3 commits:
Alter table to set the write size if the size is specified?
Do the truncate
Do the insert?